syscall.go

```diff
@@ -147,19 +158,25 @@ func waitForBlockDevice(deviceName string) error {
 		fieldLogger.Info("Started listening for udev events for block device hotplug")
 
 		// Check if the device already exists.
-		if _, err := os.Stat(devicePath); err == nil {
+		if _, err := os.Stat(path); err == nil {
 			fieldLogger.Info("Device already hotplugged, quit listening")
 		} else {
 
 			for d := range ch {
 				fieldLogger = fieldLogger.WithFields(logrus.Fields{
 					"udev-path":  d.Syspath(),
+					"dev-path":   d.Devpath(),
 					"udev-event": d.Action(),
 				})
 
 				fieldLogger.Info("got udev event")
-				if d.Action() == "add" && filepath.Base(d.Devpath()) == deviceName {
-					fieldLogger.Info("Hotplug event received")
+
+				//if isSCSIAddr && d.Action() == "add" && strings.Contains(d.Devpath(), deviceName) {
+				if isSCSIAddr && d.Action() == "add" && strings.Contains(d.Devpath(), "/0:0:"+deviceName+"/block") {
```

Comment:
Can you create a global private variable for the `0:0` (and ideally change `scsiDiskPrefix` to use it too)?